### PR TITLE
automated registration of mesh primitives (fixes #1509)

### LINKED ISSUE #1509: programmatically create geometry "primitives"
**Description:**

Right now, we manually define each geometry primitive (e.g, `<a-box>` in `extras/primitives/primitives/a-box.js`). 

When we register a geometry, we should just automatically register a corresponding primitive. Since primitives are *currently* on the `extras/` layer, we should maybe wrap `registerGeometry`.

### PR DESCRIPTION
**Pull request by ngokevin**
**Description:** Automate registration of mesh primitives. When we add a geometry, don't have to worry about creating a new primitive. Less prone to bugs, and this PR gives the new *-hedron primitives for free.

### CODE REVIEW

File: tests/extras/primitives/primitives/meshPrimitives.test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        assert.equal(el.getAttribute('material').color, 'red');
+        assert.equal(el.getAttribute('geometry').depth, 5);
+      });
+      done();
+    });
+  });
+
+  test('registers every geometry', function () {
+    function removeA (str) { return str.replace('a-', ''); }
+    console.log(Object.keys(primitives).map(removeA).map(utils.styleParser.toCamelCase));
```
Comment (ngokevin) [inline tests/extras/primitives/primitives/meshPrimitives.test.js]: rm

Comment (dmarcos) [inline tests/extras/primitives/primitives/meshPrimitives.test.js]: what's does this comment mean?

File: src/extras/primitives/primitives/meshPrimitives.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+var geometries = require('../../../core/geometry').geometries;
+var geometryNames = require('../../../core/geometry').geometryNames;
+var registerPrimitive = require('../registerPrimitive');
+var utils = require('../../../utils/');
+
+// For testing.
+var meshPrimitives = module.exports = {};
+
+// Generate primitive for each geometry type.
+geometryNames.forEach(function registerMeshPrimitive (geometryName) {
```
Comment (dmarcos) [inline src/extras/primitives/primitives/meshPrimitives.js]: What about if you register a geometry after AFRAME has loaded?

Comment (ngokevin) [inline src/extras/primitives/primitives/meshPrimitives.js]: should i do that in this PR?

File: tests/extras/primitives/primitives/meshPrimitives.test.js (hunk 1 above)
Comment (ngokevin) [inline tests/extras/primitives/primitives/meshPrimitives.test.js]: need to remove these logs

Comment (dmarcos) [inline tests/extras/primitives/primitives/meshPrimitives.test.js]: let's remove them

File: src/extras/primitives/primitives/meshPrimitives.js (hunk 2 above)
Comment (dmarcos) [inline src/extras/primitives/primitives/meshPrimitives.js]: Maybe later.

File: tests/extras/primitives/primitives/meshPrimitives.test.js (hunk 1 above)
Comment (ngokevin) [inline tests/extras/primitives/primitives/meshPrimitives.test.js]: k

**Pull request merged**